tests/handlers/InputAdded.test.ts

```diff
@@ -0,0 +1,127 @@
+    };
+});
+const tokenAddress = dataSlice(input.payload, 1, 21).toLowerCase(); // 20 bytes for address
+const from = dataSlice(input.payload, 21, 41).toLowerCase(); // 20 bytes for address
+const amount = getUint(dataSlice(input.payload, 41, 73)); // 32 bytes for uint256
+describe('InputAdded', () => {
+    let inputAdded: InputAdded;
+    let erc20;
+    const mockTokenStorage = new Map();
+    const mockDepositStorage = new Map();
+    const mockInputStorage = new Map();
+    const mockApplicationStorage = new Map();
+    beforeEach(() => {
+        inputAdded = new InputAdded(
+            mockTokenStorage,
+            mockDepositStorage,
+            mockApplicationStorage,
+            mockInputStorage,
+        );
+        erc20 = new Contract(ctx, block.header, tokenAddress);
+        mockTokenStorage.clear();
+        mockDepositStorage.clear();
+        mockApplicationStorage.clear();
+        mockInputStorage.clear();
+        vi.clearAllMocks();
+    });
+    describe('handlePayload(log)', async () => {
+        test('call with the correct params', async () => {
+            vi.spyOn(inputAdded, 'handlePayload');
+            inputAdded.handlePayload(input, block, ctx);
+            expect(inputAdded.handlePayload).toHaveBeenCalledWith(
+                input,
+                block,
+                ctx,
+            );
+        });
+        test('call the ERC20 module', async () => {
+            await inputAdded.handlePayload(input, block, ctx);
+            expect(erc20.name).toBeCalledTimes(1);
+            expect(erc20.symbol).toBeCalledTimes(1);
+            expect(erc20.decimals).toBeCalledTimes(1);
+        });
+        test('return the correct deposit value', async () => {
+            const name = 'SimpleERC20';
+            const symbol = 'SIM20';
+            const decimals = 18;
+            const token = new Token({
+                id: tokenAddress,
+                name,
+                symbol,
+                decimals,
+            });
+            const deposit = new Erc20Deposit({
+                id: input.id,
+                amount,
+                from,
+                token,
+            });
+            erc20.name.mockResolvedValueOnce('SimpleERC20');
+            erc20.symbol.mockResolvedValue('SIM20');
+            erc20.decimals.mockResolvedValue(18);
+            const handlePayload = await inputAdded.handlePayload(
+                input,
+                block,
+                ctx,
+            );
+            expect(handlePayload).toStrictEqual(deposit);
+        });
+        test('msgSender is not the correct ERC20PortalAddress', async () => {
+            input.msgSender = '0x42985af528673AF020811c339Bf62497160Fe087';
+            const handlePayload = await inputAdded.handlePayload(
+                input,
+                block,
+                ctx,
+            );
+            expect(handlePayload).toBe(undefined);
+        });
+    });
+    describe('handle', async () => {
```

Comment:
Hi @dandheedge, 
I think a case is missing for the `msgSender` to be the ERC20Portal address where a deposit would be created and need to be checked. If you run the `ci:test` that generates the coverage, you will see what I mean by "missing case".